### ISSUE TITLE
Issues with Hybrid setups (WMR/Quest with Index Controllers) and chaperone

### ISSUE THREAD

**Issue by spoonyspork** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#426]
**Describe the bug**
This bug is a bit difficult to describe. I use Index controllers with a WMR HMD and a Quest HMD. In the past, if I went to the 'chaperone' section of Advanced Settings and clicked 'reset to disk', the chaperone outline that would be saved would be the chaperone of the current HMD. I could then set a profile using this chaperone and swap between them as I swapped HMD's.

Now there are two issues:

* Clicking 'reset to disk' ignores the current chaperone and all I see when I click 'show playspace' is a small square. However after a bit of experimenting, I noticed when I set haptic proximity warning, it was actually using the Quest's playspace. However the chaperone bounds wouldn't show up, which lead to:

* On swapping to WMR, the same thing happened, except I noticed it was still using the Quest's boundaries. However, the chaperone bounds would show up (with the small box still in the center of the playspace) when the controllers would hit the edge of those boundaries. The Quest boundary was also still lined up with... would it be called the 'world' space of the quest? So the space was tilted and forward, offset from the real life space.

This happens whether or not I start with the option in advanced settings to allow external apps to change the playspace.

**To Reproduce**
1. Start steamVR with a non-Index HMD, calibrate the space with space calibrator.
2. Go into advanced settings and attempt to create a profile using the world space of the HMD

**Expected behavior**
Advanced settings should accept the boundaries from the given HMD and allow separate profiles.

**Screenshots**
Here is a link with a video that might more obviously show the problem: https://www.youtube.com/watch?v=ueazO_LAm7k&feature=youtu.be

**Software (please complete the following information):**
 - OS: Windows 10 1909
 - Advanced Settings Version: 5.0.2

**Hardware (please complete the following information):**
 - CPU: 7700k
 - Graphics Card: 1070ti
 - HMD: Odyssey + & Quest

**Additional context**
Again this is difficult to describe, as I am not sure exactly how it worked from the beginning. But I have always been able to get the chaperone in advanced settings to match the chaperone of the given HMD, and have never had that small square in the middle (note it is NOT the playspace center marker). This is needed because the chaperone is disabled for the Index controllers when a non-matching HMD is used. The chaperone profile allows a haptic wall and visual floor chaperone, and also must be running in order to play games in Revive (otherwise the game doesn't seem to know where to put the player and throws you somewhere random in the map). I can provide another video showing the setup from start to finish if need be.

**Comment by spoonyspork** [comment]
This appears to be an issue with the way advanced settings interacts with other OpenVR apps (or rather the way other OpenVR apps interact with Advanced settings?) In this case it was the play area of virtual desktop being used as soon as the Index controllers were turned on. Fixed by deleting every folder having to do with openVR I could find, and reinstalling everything one at a time. At the point space calibrator is installed, the chaperone bounds need to be pasted in before the index controllers are turned on... otherwise the Quest/Virtual Desktop playspace was being pulled in (from where, I would like to know). After this, advanced settings can be installed. Clicking 'reload from disk' in the chaperone section then pulls in the bounds from WMR/space calibrator and everything can be set as desired and a profile saved.

I am not sure if this would be considered a bug or feature request, and if it would be this for Advanced Settings, Space Calibrator, or Virtual Desktop. Going to go ahead and close it here since it's working as previously expected.

**Comment by ykeara** [comment]
Hate to say it but you have a complicated set-up XD.

Right so Steam's chaperone is saved in Chaperone.vrchap I think in %steaminstall%/config.

Can't speak where Quests is saved especially when through virtual desktop (possible on quest, possible somewhere related to how virtual desktop handles it)

Now onto how OVRAS handles chaperone.

We ARE fairly aggressive in how we hold onto the chaperone, to my knowledge we are the only app (perhaps other than room-setup) that uses the `working` state of the chaperone. We do this because we assume it should have better performance and less disk writes, as well as it not being as limited in terms of play area size. ~50 km vs ~8km.

We also maintain an `offset` to the chaperone we are using.... this is primarily done as we have 0 error accumulation in our chaperone movement.

So we are a bit of a `bully` in what other chaperones can do while we are active XD. we do explicitly allow room-setup.

The biggest thing that you should do if you are applying other chaperone's while We are active is have "allow external chaperone edits" in settings checked, and before you apply.... use reset in offsets or fix floor in space fix page. [and then don't move w/ OVRAS until you apply your changes] This should help you.

Also in your-setup you may want to disable chaperone auto-recovery in settings.... or at least watch for the warning to see if its an issue in logs.

tl;dr its complicated, but I think "which" chaperone is loading from disk shouldn't be dependent on us.... (though if our error recovery kicks in because bad-shutdown it will load what we have saved) So order matters, I'm not sure its really a bug on anybodies end, but you have essentially 2.5 chaperone systems, and 3 apps that play with them. It's not really feasible for us to add edge of edge cases into our program either. It just makes it crazy complicated. glad you figured it out.

**Comment by spoonyspork** [comment]
Thank you for the reply!

Yeah my setup is a bit strange, though no more strange than anyone using full body tracking I think! Pretty sure the actual issue was me doing *something* out of order when adding the Quest to the mix (I think every time I went into SteamVR with the Quest it was saving a new playspace and then space calibrator and advanced settings started fighting over who got to set the bounds... and then I continued doing things out of order, like when to tell space calibrator to get the bounds and when to tell advanced settings to let space calibrator get the bounds with 'allow external chaperone edits')

I deleted chaperone.vrchap completely and it affected nothing at all, and the file wasn't recreated on reinstall of any of the openVR apps which I thought was pretty weird! Viewing the actual content, it had a date stamp of July 2019 which was the first time I used a hybrid setup...

I am happy it was just me being a dolt, but it's going to keep bothering me where the heck any of the apps were even finding the Quest's/Virtual Desktop's chaperone(s)! What finally clued me in was going into Steam via VD again and then having a *4th* playspace outline pop up the next time I went in with WMR... XD
